### PR TITLE
Special case TestOpenFileLinks on macOS

### PR DESCRIPTION
**Pull request by xacrimon**
Currently `TestOpenFileLinks` will fail on macOS sometimes due to the `t.TempDir` function not returning a path that allows symlinks (macOS is weird with it's special folders/volumes/paths).

This PR special cases macOS to use another common temp storage folder that allow the test to complete.

### CODE REVIEW

**Review by gzdunek — APPROVED** [review]

File: lib/utils/fs_test.go [hunk 1]
```diff
@@ -38,7 +40,32 @@ func TestOpenFileLinks(t *testing.T) {
...
-	rootDir := t.TempDir()
+
+	// macOS is special cased here since t.TempDir() returns a path to a protected directory that doesn't allow symlinks.
+	var rootDir string
+	switch runtime.GOOS {
+	case "darwin":
+		suffix := make([]byte, 8)
+		_, err := rand.Read(suffix)
+		if err != nil {
+			t.Fatalf("failed to generate random suffix: %v", err)
```
Comment (zmb3) [inline lib/utils/fs_test.go:51]: Should we use `require.NoError` for consistency with all of our other tests?

File: lib/utils/fs_test.go [hunk 2]
```diff
@@ -38,7 +40,32 @@ func TestOpenFileLinks(t *testing.T) {
...
+		}
+
+		rootDir = "/private/tmp/" + "teleport-test-" + hex.EncodeToString(suffix)
+		err = os.Mkdir(rootDir, 0700)
```
Comment (zmb3) [inline lib/utils/fs_test.go:55]: I think you can replace all of this with

```
os.MkdirTemp("/private/tmp/teleport-test-*")
```

File: lib/utils/fs_test.go (hunk 1 above)
Comment (xacrimon) [reply lib/utils/fs_test.go:51]: fixed

File: lib/utils/fs_test.go (hunk 2 above)
Comment (xacrimon) [reply lib/utils/fs_test.go:55]: thanks, didn't know that existed

Comment (xacrimon) [reply lib/utils/fs_test.go:55]: Fixed

Comment (zmb3) [reply lib/utils/fs_test.go:55]: You don't need to generate a random suffix anymore though. That's what MkdirTemp will do for you.

File: lib/utils/fs_test.go (hunk 1 above)
Comment (zmb3) [reply lib/utils/fs_test.go:51]: Where? I'm not seeing the fix?

File: lib/utils/fs_test.go (hunk 2 above)
Comment (xacrimon) [reply lib/utils/fs_test.go:55]: Just forgot to push

Comment (xacrimon) [reply lib/utils/fs_test.go]: Done

File: lib/utils/fs_test.go [hunk 3]
```diff
@@ -38,10 +38,29 @@ func TestOpenFileLinks(t *testing.T) {
...
 	// hardfile-h -> hardfile
-	rootDir := t.TempDir()
+
+	// macOS is special cased here since t.TempDir() returns a path to a protected directory that doesn't allow symlinks.
+	var rootDir string
+	var err error
+	switch runtime.GOOS {
+	case "darwin":
+		rootDir, err = os.MkdirTemp("/private/tmp", "teleport-test-*")
+		if err != nil {
```
Comment (zmb3) [inline lib/utils/fs_test.go:48]: No need for the if statement anymore.

File: lib/utils/fs_test.go [hunk 4]
```diff
@@ -38,10 +38,29 @@ func TestOpenFileLinks(t *testing.T) {
...
+			require.NoError(t, err)
+		}
+
+		t.Cleanup(func() {
+			err := os.RemoveAll(rootDir)
+			if err != nil {
+				require.NoError(t, err)
```
Comment (zmb3) [inline lib/utils/fs_test.go:55]: Use require here too.

File: lib/utils/fs_test.go (hunk 3 above)
Comment (xacrimon) [reply lib/utils/fs_test.go]: Ah sorry, dumb mistake, it's late. Time to head out. Fixed

**Review by zmb3 — APPROVED** [review]
Thanks for your patience :-)

**Comment by xacrimon** [comment]
THanks for the speed reviews @zmb3. They're unmatched.

**Pull request merged**